README.md

```diff
@@ -62,13 +62,36 @@ installed system wide. To do this:
 
 ### macOS (OS X)
 
-If you're using macOS (formerly OS X), `/etc/bashrc` is apparently not sourced at
-all. In that case, you can put the `bash_completion` file in `/sw/etc`
-and add the following code to `~/.bash_profile`:
+If you're using macOS (formerly OS X), `/etc/bashrc` is apparently not sourced
+at all, and `~/.bashrc` is not sourced from `~/.bash_profile` by default
+(because `~/.bash_profile` is not created by default).  In this case, the
+standard way is to configure `~/.bash_profile` to source `~/.bashrc` and write
+interactive settings in `~/.bashrc`.  You can source `~/.bashrc` in
+`~/.bash_profile` in the following way:
 
 ```shell
-if [ -f /sw/etc/bash_completion ]; then
-   . /sw/etc/bash_completion
+# ~/.bash_profile
+
+if [[ -f ~/.bashrc ]]; then
+  source ~/.bashrc
+fi
+```
+
+Then, you can source `bash-completion` in your `~/.bashrc`.  It should be noted
+that `bash-completion` should not be sourced in `~/.bash_profile` because
+`~/.bash_profile` is only sourced in interactive login shell sessions.  If you
+start nested Bash sessions, the interactive settings in `~/.bash_profile` will
+disappear.  It is strongly recommended to source `~/.bashrc` from
+`~/.bash_profile` and write interactive settings in `~/.bashrc`.
+
+For example, if you install `bash-completion` using Homebrew, it will install
+the entry point of `bash-completion` to
+`$HOMEBREW_PREFIX/etc/profile.d/bash_completion.sh`.  We can source it by
+adding the following to our startup file `~/.bashrc`:
```

Comment:
Doesn't homebrew do enablement of bash-completion by default out of the box?

If not, do they have instructions of their own how to do it?

If they do, I'd rather not ship specific documentation of our own (if it's redundant) but rather point to theirs, and if it's not redundant, I feel it would be good to somehow explicitly acknowledge that the package does come with docs of its own, and we recommend the user to look into that as well (or recommend against something in it, whichever way it is).

Reply:
Actually, this is [explained in Homebrew's `bash-completion`](https://github.com/Homebrew/homebrew-core/blob/5df158e314883cda5ff0f9c258d289af64c712db/Formula/b/bash-completion%402.rb#L49-L54). I think this string will be presented to the user on the installation time. On the other hand, they don't describe it on [its web page](https://formulae.brew.sh/formula/bash-completion).

One non-trivial thing is that the `.sh` files in `/etc/profile.d` are intended to be automatically sourced by a system-wide shell configuration. Then, one would naively expect Homebrew to configure `/etc/profile.d/*.sh` to be automatically sourced so that a user can simply source a Homebrew shell configuration file (e.g. at `$HOMEBREW_PREFIX/etc/bashrc` or `$HOMEBREW_PREFIX/etc/profile`). I don't have macOS, but my guess is that Homebrew doesn't offer such a configuration, so a user needs to manually source each file in `$HOMEBREW_PREFIX/ext/profile.d`.

Reply:
> so a user needs to manually source each file in $HOMEBREW_PREFIX/ext/profile.d.

This is the case.